client/consul.go

```diff
@@ -336,3 +338,9 @@ func (c *ConsulService) makeCheck(service *structs.Service, check *structs.Servi
 	}
 	return cr
 }
+
+func (c *ConsulService) printLogMessage(message string, v ...interface{}) {
+	if _, ok := c.node.Attributes["consul.version"]; ok {
```

Comment:
I still don't follow. There are two scenarios:

1. Consul failed temporarily. Let's suppose consul initially is missing but will show up. Nomad starts first. Then Consul starts 20 seconds later. The second fingerprinter pass will enable the consul functionality. From here on any failure to sync a service with Consul is likely an operational problem, so we should show the logs.

2. Consul is not running and never will be. In this case the fingerprinter detects Consul is not there, and in fact Consul never shows up. Why would we ever call `NewConsulService` at all, or why would it not immediately no-op? Showing an info log saying we're registering a service when we in fact will never register the service is spurious, and the entire feature seems extraneous.

If we have to have the service registered permanently when the task starts _just in case consul shows up some day_, it seems like we should still be able to do this check much earlier and skip all of this additional complexity with respect to filtering the logs and retrying something that's never going to succeed.

For example in [`performSync`](https://github.com/hashicorp/nomad/blob/r-mute-consul-debug-messages/client/consul.go#L196-L197) if we can't connect to consul we can just skip the entire run, log an `ERR` message that we can't connect, and move on.